### DISCUSSION TITLE
Show LSP info inside the window

### DISCUSSION THREAD

**Discussion by serhez** [soulis-1256/eagle.nvim#13] [General]
Hello, could it also be possible to trigger `vim.lsp.buf.hover()` on mouse hover?

**Comment by soulis-1256** [comment]
@serhez Not right now, but it will be in the next update. I'm aware.

**Comment by soulis-1256** [comment]
Hi, I want to inform you and anyone that is interested that I finally reached the final stage of containing lsp info inside the eagle window. It's all in the new_features branch and it's pretty much done even as I write this (if you really want to use it right now, use ```git checkout new_features```). After I improve/test it some more, a big update containing this and some optimizations will be dropped on the main branch and then I will close this issue. Here is the result as of right now:

![image](https://github.com/soulis-1256/eagle.nvim/assets/118274635/49265fc1-a037-464f-939d-5df269c3cba3)

**Comment by soulis-1256** [comment]
Instead of closing the issue once I merge new_features with main, I am converting this issue to a discussion, to have a place for future questions/ideas about this feature (ie how it works, how it can be improved/optimized).

**Comment by serhez** [comment]
Hello, looks amazing! I'll be testing it this week. One thing I noticed was that, if I have a diagnostic + the token is hoverable via LSP, both the diagnostic and the LSP hover information are shown in a single floating window, which seems also unformatted (no colours, etc.). Do you think it would be better for diagnostics to take preference over LSP hover?

**Reply by serhez** [reply]
If you want to test it: in Python with Pyright, you can import some package that is not installed and you will be able to have both a warning and also inspect the token via hover.

**Reply by soulis-1256** [reply]
Hi, if I understand correctly, you are telling me there is a case where the LSP Server (in your case Pyright), will have knowledge of the imported package and will show information, despite the package not being actually installed in the system. In that case we get a warning as a Diagnostic that is displayed in the beginning of the window, and then after that there is the information about the package from the LSP Server. I suppose you mean this is contradictory, that if the package is not installed, then the LSP Info should not be shown at all. Am I correct thus far?

**Reply by serhez** [reply]
The information about the package is not what's shown in the floating window, but rather some generic information about importing a package (i.e., pyright tells you how `import` works). Perhaps the example I gave was not good because there should not be hover information in that case. What I mean is that, whenever there are both LSP hover and also diagnostics available, the floating window seems to reflect both. I'm suggesting maybe it would be better to only show one of them based on preference, or perhaps 2 different floating windows?

**Reply by serhez** [reply]
Also (and it may be related to the above), it seems that information displayed when hovering over LSP tokens sticks to the floating window: when I hover over a new token I can see the previous token's information appended to the new token's information. I'm seeing this using Pyright.

**Reply by soulis-1256** [reply]
About the last thing you said about lsp info being appended, which is more important than the first: have you done a ```git pull``` to get my latest commits? I installed Pyright myself using Mason and I can't reproduce.
Edit: about the other thing, a screenshot showing the contents of the eagle window would be ideal

**Reply by soulis-1256** [reply]
I just pushed a commit to new_features that may have fixed the problem you saw, where lsp info would appear in wrong places. Tell me if that's the case when you can. Now, for the diagnostics and lsp info saying the same thing, as I said in my previous reply, I will need a screenshot to see what you mean exactly.

**Reply by MuntasirSZN** [reply]
@soulis-1256 Is this implemented?

**Reply by soulis-1256** [reply]
> @soulis-1256 Is this implemented? 

Hi, if you mean showing LSP info, yes it has been implemented for some time now. I never thought to close this discussion.

**Reply by MuntasirSZN** [reply]
Oh, ok!
